component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -555,4 +557,76 @@ public static boolean isEventHandlerBasedEmailSenderEnabled() {
 				.get(TOTPAuthenticatorConstants.USE_EVENT_HANDLER_BASED_EMAIL_SENDER);
 		return Boolean.parseBoolean(eventHandlerBasedEmailSenderProperty);
 	}
+
+	/**
+	 *  Get Account Lock Connector Configs.
+	 *
+	 * @param tenantDomain Tenant domain.
+	 * @return Account Lock Connector Configs.
+	 * @throws AuthenticationFailedException Exception on authentication failure.
+	 */
+	public static Property[] getAccountLockConnectorConfigs(String tenantDomain) throws AuthenticationFailedException {
+
+		Property[] connectorConfigs;
+		try {
+			connectorConfigs = TOTPDataHolder.getInstance()
+					.getIdentityGovernanceService()
+					.getConfiguration(
+							new String[]{
+									TOTPAuthenticatorConstants.PROPERTY_ACCOUNT_LOCK_ON_FAILURE,
+									TOTPAuthenticatorConstants.PROPERTY_ACCOUNT_LOCK_ON_FAILURE_MAX,
+									TOTPAuthenticatorConstants.PROPERTY_ACCOUNT_LOCK_TIME,
+									TOTPAuthenticatorConstants.PROPERTY_LOGIN_FAIL_TIMEOUT_RATIO
+							}, tenantDomain);
+		} catch (Exception e) {
+			throw new AuthenticationFailedException(
+					"Error occurred while retrieving account lock connector configuration", e);
+		}
+		return connectorConfigs;
+	}
+
+	/**
+	 * Check whether account locking is enabled for TOTP.
+	 *
+	 * @return True if account locking is enabled for TOTP.
+	 */
+	public static boolean isAccountLockingEnabledForTotp() {
+
+		return Boolean.parseBoolean(
+				getTOTPParameters().get(TOTPAuthenticatorConstants.ENABLE_ACCOUNT_LOCKING_FOR_FAILED_ATTEMPTS));
+	}
+
+	public static boolean isAccountLocked(String userName, String tenantDomain, String userStoreDomain)
+			throws AuthenticationFailedException {
+
+		try {
+			return TOTPDataHolder.getInstance().getAccountLockService()
+					.isAccountLocked(userName, tenantDomain, userStoreDomain);
+		} catch (Exception e) {
```

Comment:
Is there any specific reason here to catch Exception and throw a subclass of Exception?

Reply:
`processAuthenticationResponse` method only throws `AuthenticationFailedException` to handle in `AbstractApplicationAuthenticator`. This method is invoked inside `processAuthenticationResponse`

Reply:
Fixed  https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/73/commits/7855d43085638b13d6de92a0460ba4267a20bb9e